File: GCP/examhandler/network.go
// network.go
package examhandler

import (
    "net/http"
    "go.mongodb.org/mongo-driver/mongo"
    "go.mongodb.org/mongo-driver/bson"
)

func handleNetworkQuestions(w http.ResponseWriter, collection *mongo.Collection, pathParts []string) {
    if len(pathParts) < 3 {
        sendError(w, "Exam date not specified", http.StatusBadRequest)
        return
    }
    handlePdfQuestions(w, collection)
}

func handleNetworkAnswers(userAnswers map[string]string, collection *mongo.Collection, examDate string) ([]QuizResult, int) {
    return handlePdfAnswers(userAnswers, collection, examDate,
        func(q bson.M, date string) (int, bool) {
            answers := q["answers"].(bson.M)
            var answer int
            switch date {
            case "20240825":
                answer = int(answers["20240825"].(int32))
            case "20240519":
                answer = int(answers["20240519"].(int32))
            case "20240225":
                answer = int(answers["20240225"].(int32))
            case "20231105":
                answer = int(answers["20231105"].(int32))
            case "20230820":
                answer = int(answers["20230820"].(int32))
            case "20230521":
                answer = int(answers["20230521"].(int32))
            case "20230226":
                answer = int(answers["20230226"].(int32))
            default:
                return 0, false
            }
            return answer, true
    })
}

File: lambda/network.py
from base_handler import get_pdf_questions
from pdf_handler import handle_pdf_answers

def handle_network_questions(collection):
    return get_pdf_questions(collection)

def handle_network_answers(user_answers, collection, exam_date):
    def get_answer(question, date):
        answers = question.get('answers', {})
        answer_map = {
            '20240825': answers.get('20240825'),
            '20240519': answers.get('20240519'),
            '20240225': answers.get('20240225'),
            '20231105': answers.get('20231105'),
            '20230820': answers.get('20230820'),
            '20230521': answers.get('20230521'),
            '20230226': answers.get('20230226')
        }
        answer = answer_map.get(date)
        return answer, answer is not None
    
    return handle_pdf_answers(user_answers, collection, exam_date, get_answer)
